E0076 needs to be updated to new format

From: src/test/compile-fail/E0076.rs

Error E0076 needs a span_label, updating it from:

```
error[E0076]: SIMD vector should be homogeneous
  --> src/test/compile-fail/E0076.rs:14:1
   |
14 | struct Bad(u16, u32, u32); //~ ERROR E0076
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0076]: SIMD vector should be homogeneous
  --> src/test/compile-fail/E0076.rs:14:1
   |
14 | struct Bad(u16, u32, u32); //~ ERROR E0076
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ SIMD elements must have the same type
```

Bonus: The span should focus on the arguments

```
error[E0076]: SIMD vector should be homogeneous
  --> src/test/compile-fail/E0076.rs:14:1
   |
14 | struct Bad(u16, u32, u32); //~ ERROR E0076
   |            ^^^^^^^^^^^^^ SIMD elements must have the same type
```

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -1228,7 +1228,9 @@ pub fn check_simd<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, sp: Span, id: ast::Node
             }
             let e = fields[0].ty(tcx, substs);
             if !fields.iter().all(|f| f.ty(tcx, substs) == e) {
-                span_err!(tcx.sess, sp, E0076, "SIMD vector should be homogeneous");
+                struct_span_err!(tcx.sess, sp, E0076, "SIMD vector should be homogeneous")
+                                .span_label(sp, &format!("SIMD elements must have the same type"))
+                                .emit();
                 return;
             }
             match e.sty {

--- a/src/test/compile-fail/E0076.rs
+++ b/src/test/compile-fail/E0076.rs
@@ -11,7 +11,9 @@
 #![feature(repr_simd)]
 
 #[repr(simd)]
-struct Bad(u16, u32, u32); //~ ERROR E0076
+struct Bad(u16, u32, u32);
+//~^ ERROR E0076
+//~| NOTE SIMD elements must have the same type
 
 fn main() {
 }

